src/org/labkey/response/ResponseController.java

```diff
@@ -1203,4 +1216,165 @@ public void setDescription(String description)
+        public static final String WCP_USERNAME = "wcpUsername";
+        public static final String WCP_PASSWORD = "wcpPassword";
+
+        @Override
+        public ModelAndView getView(ServerConfigurationForm form, boolean reshow, BindException errors) throws Exception
+        {
+            JspView<ServerConfigurationForm> view = new JspView<>("/org/labkey/response/view/responseServerConfiguration.jsp", form, errors);
+            view.setTitle("Global Settings");
+            return view;
+        }
+
+        @Override
+        public boolean handlePost(ServerConfigurationForm form, BindException errors) throws Exception
+        {
+            if (form.getMetadataLoadLocation() != null && form.getMetadataLoadLocation().equals(FILE) && !Files.exists(Paths.get(form.getMetadataDirectory())))
+            {
+                errors.addError(new LabKeyError("Metadata Directory path is invalid"));
+            }
+            else if (form.getMetadataLoadLocation() != null && form.getMetadataLoadLocation().equals(WCP_SERVER))
+            {
+                if (form.getWcpUsername() == null || form.getWcpUsername().isBlank())
+                    errors.addError(new LabKeyError("WCP username must not be blank"));
+
+                if (form.getWcpPassword() == null || form.getWcpPassword().isBlank())
+                    errors.addError(new LabKeyError("WCP password must not be blank"));
+
+                if (form.getWcpBaseURL() == null || !form.getWcpBaseURL().toUpperCase().matches("^(HTTP|HTTPS)://.*$"))
+                    errors.addError(new LabKeyError("WCP Base URL must begin with 'http://' or 'https://'"));
+
+                if (form.getWcpBaseURL() == null || !form.getWcpBaseURL().endsWith("/StudyMetaData"))
+                    errors.addError(new LabKeyError("WCP Base URL must end with '/StudyMetaData'"));
+            }
+
+            if (errors.getErrorCount() > 0)
+                return false;
+            else
+            {
+                PropertyManager.PropertyMap props = PropertyManager.getEncryptedStore().getWritableProperties(getContainer(), RESPONSE_SERVER_CONFIGURATION, true);
+                Map<String, String> valuesToPersist = form.getOptions();
+
+                if (!valuesToPersist.isEmpty())
+                {
+                    props.putAll(valuesToPersist);
+                    props.save();
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        @Override
+        public URLHelper getSuccessURL(ServerConfigurationForm serverConfigurationForm)
+        {
+            return urlProvider(AdminUrls.class).getAdminConsoleURL();
+        }
+
+        @Override
+        public void validateCommand(ServerConfigurationForm target, Errors errors)
+        {
+
+        }
+
+        @Override
+        public void addNavTrail(NavTree root)
+        {
+            setHelpTopic("TODO"); // Rosaline TODO
```

Comment:
Since the page in question is `https://www.labkey.org/FDAMyStudiesHelp/wiki-page.view?name=setup`, which has a different URL/is in a portal compared to most doc pages, should we be attempting to set this here?

Reply:
I would suggest creating a special subclass of `HelpTopic` (maybe call it `MyStudiesHelpTopic`) that overrides the `getHelpTopicHref()` method. Then we can simply pass in a MyStudiesHelpTopic to `setHelpTopic()`.